internal/restic/counted_blob_set.go

```diff
@@ -0,0 +1,66 @@
+package restic
+
+import "sort"
+
+// CountedBlobSet is a set of blobs.
+type CountedBlobSet map[BlobHandle]uint8
+
+// NewCountedBlobSet returns a new CountedBlobSet, populated with ids.
+func NewCountedBlobSet(handles ...BlobHandle) CountedBlobSet {
+	m := make(CountedBlobSet)
+	for _, h := range handles {
+		m[h] = 0
+	}
+
+	return m
+}
+
+// Has returns true iff id is contained in the set.
+func (s CountedBlobSet) Has(h BlobHandle) bool {
+	_, ok := s[h]
+	return ok
+}
+
+// Insert adds id to the set.
+func (s CountedBlobSet) Insert(h BlobHandle) {
+	s[h] = 0
```

Comment:
Shouldn't this be
```suggestion
	s[h] = 1
```

Reply:
No. `0` is correct here. The check, whether a blob is part of the set does not use the value. In fact the value stored in the CountedBlobSet is not for use by the `CountedBlobSet`. Instead it is exclusively used to count how many times a blob is contained in the index, which happens here https://github.com/restic/restic/pull/3899/files#diff-33a0be5385e73907c2b87dc06a9ebc1822c4abe53738ed482dd152e26c86cac5R308 .